sds/src/match_validation/aws_validator.rs

```diff
@@ -1,22 +1,145 @@
+use std::fmt;
+
+use ahash::AHashMap;
 use async_trait::async_trait;
+use futures::future::join_all;
+use reqwest::Client;
 
-use crate::{CompiledRuleDyn, RuleMatch};
+use crate::{CompiledRuleDyn, MatchStatus, RuleMatch};
 
-use super::match_validator::MatchValidator;
+use super::{
+    config::{AwsConfig, AwsType, MatchValidationType},
+    match_validator::MatchValidator,
+    validator_utils::generate_aws_headers_and_body,
+};
 
 pub struct AwsValidator {
-    // Otherwise struct is not allocated by compiler opt
-    // will be removed in next PR
-    _useless: bool,
+    pub config: AwsConfig,
 }
 
 impl AwsValidator {
-    pub fn new() -> Self {
-        AwsValidator { _useless: false }
+    pub fn new(config: AwsConfig) -> Self {
+        AwsValidator { config }
     }
 }
 
 #[async_trait]
 impl MatchValidator for AwsValidator {
-    async fn validate(&self, _: &mut Vec<RuleMatch>, _: &Vec<Box<dyn CompiledRuleDyn>>) {}
+    async fn validate(
+        &self,
+        matches: &mut Vec<RuleMatch>,
+        scanner_rules: &Vec<Box<dyn CompiledRuleDyn>>,
+    ) {
+        // Let's regroup matches per type
+        let mut aws_id_matches_idx = vec![];
+        let mut aws_secret_matches_idx = vec![];
+
+        for (idx, m) in matches.iter().enumerate() {
+            let rule: &Box<dyn CompiledRuleDyn> = &scanner_rules[m.rule_index];
+            match rule.get_match_validation_type() {
+                Some(MatchValidationType::Aws(AwsType::AwsId)) => {
+                    aws_id_matches_idx.push(idx);
+                }
+                Some(MatchValidationType::Aws(AwsType::AwsSecret(_))) => {
+                    aws_secret_matches_idx.push(idx);
+                }
+                _ => {}
+            }
+        }
+
+        let mut match_status_per_pairs_of_matches_idx: AHashMap<(usize, usize), MatchStatus> =
+            AHashMap::new();
+        for id_index in 0..aws_id_matches_idx.len() {
+            for secret_index in 0..aws_secret_matches_idx.len() {
+                match_status_per_pairs_of_matches_idx.insert(
+                    (
+                        aws_id_matches_idx[id_index],
+                        aws_secret_matches_idx[secret_index],
+                    ),
+                    MatchStatus::NotChecked,
+                );
+            }
+        }
+
+        let client = Client::new();
+        // Let's try all combination of aws_id and aws_secret
+        let futures = match_status_per_pairs_of_matches_idx.iter_mut().map(
+            |((id_index, secret_index), match_status)| {
+                let client = client.clone();
+                let match_id = &matches[*id_index];
+                let match_secret = &matches[*secret_index];
+                async move {
```

Comment:
The `async` stuff is fine here, just wanted to let you know that `reqwest` does have a sync variant (different import) if for some reason we find the async is to much work / not enough benefit.